### PR TITLE
Add namespaces to full marc-xml

### PR DESCRIPTION
**Pull request by billdueber**
This actually does three things:
  * Remove the local `marc::fastxmlwriter`, which was superseded
    by the gem of the same name (already in the gemfile) long ago
  * Include the namespace when generating XML with fastxmlwriter
  * Gate record-redirect computation with `ENV['NO_REDIRECTS']`, for convenience. 

TODO: Create a generic NO_EXTERNAL env variable.

### CODE REVIEW

**Review by aelkiss — APPROVED** [review]
Substantive changes look minimal enough; Changes look pretty minimal. I think before in dev I had just used `REDIRECT_FILE=/dev/null` but the explicit `NO_REDIRECTS` is definitely preferable.

File: indexers/common.rb [hunk 1]
```diff
@@ -81,7 +84,7 @@ def filing_version(field, str, spec)
 # to_field 'fullrecord', macr4j_as_xml
 
 to_field 'fullrecord' do |rec, acc|
-  acc << MARC::FastXMLWriter.single_record_document(rec)
+  acc << MARC::FastXMLWriter.single_record_document(rec, include_namespace: true)
```
Comment (aelkiss) [inline indexers/common.rb:87]: Actually, in testing this, the root element looks like:

```xml
<collection 
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xmlns:marc="http://www.loc.gov/MARC21/slim"
  xsi:schemaLocation="http://www.loc.gov/MARC21/slim http://www.loc.gov/standards/marcxml/schema/MARC21slim.xsd">
```

which is still not valid -- we either want it to look like `<marc:collection ...>` (and likewise all other elements) or `<collection xmlns="http://www.loc.gov/MARC21/slim">`. The former might be preferable to the extent it makes it easier to embed into other documents..

**Review by aelkiss — CHANGES_REQUESTED** [review]

File: indexers/common.rb (hunk 1 above)
Comment (aelkiss) [reply indexers/common.rb:87]: In particular this line doesn't look right - https://github.com/billdueber/marc-fastxmlwriter/blob/master/lib/marc/fastxmlwriter.rb#L28 -- this one does https://github.com/billdueber/marc-fastxmlwriter/blob/master/lib/marc/fastxmlwriter.rb#L12 but `OPEN_COLLECTION_NAMESPACE` isn't referenced anywhere

Comment (billdueber) [reply indexers/common.rb:87]: TIL that none of the marc parsers given a good goddamn about correct namespace declarations. Taking a look.

Comment (billdueber) [reply indexers/common.rb:87]: I _think_ what I've got is actually equivalent to the latter, but with extra info about what type of schema and where it's located. So, more than we need, but not incorrect. See https://stackoverflow.com/questions/34202967/xmlns-xmlnsxsi-xsischemalocation-and-targetnamespace

Comment (billdueber) [reply indexers/common.rb:87]: Fixed `MARC::FastXMLWriter` to correctly include namespaces and added better tests to it, and updated this gemfile to require `"marc-fastxmlwriter", "~>1.2"`

**Review by aelkiss — APPROVED** [review]
Verified this now generates MARC-XML with namespaces and schema declarations, and that that MARC-XML validates with Xerces.

**Pull request merged**